src/main/webapp/app/tasks/local-task.service.ts

```diff
@@ -10,8 +10,8 @@ export class LocalTaskService implements TaskService {
 
   private static readonly STORAGE_KEY: string = 'tiny.tasks';
 
-  getAll(): Observable<Task[]> {
-    return of(this.readTasks());
+  getAll(query?:string): Observable<Task[]> {
+    return of(this.readTasks(query));
```

Comment:
like stated above, the main reason to add a new function for search functionality is, that it is a bit shady to rely on an optional query parameter in my opinion and doesn't make for a very good (as in readable and maintainable) code base, but more on that topic down below